natiami/assignment3/Grid.cpp

```diff
@@ -0,0 +1,64 @@
+#include"Grid.h"
+
+Grid::Grid(){
+	this->cells = vector<vector<char> >();
+}
+
+Grid::Grid(int width, int height){
+	this->width = width;
+	this->height = height;
+	this->cells = vector<vector<char> >(height, vector<char>(width));
+	this->moves = { 0, -1, -1, 0, 0, 1, 1, 0, -1, -1, -1, 1, 1, -1, 1, 1 };

```

Comment:
Tip:

This flat list of integers can be to follow, especially if you want to update it. It is actually a list of pairs! So for this you can use the std::pair class:
```C++
std::vector<std::pair<int, int>> moves;
// Then to initialize the moves you can still use braces initialization:
moves = { {0, -1}, {-1, 0}, {0, 1}, {1,0} ...};  // The intent is clearer.
// Then the iteration is easier. No need to deal with iterating by jumps of 2 etc.
for (const std::pair<int, int>& move : moves) {
   int next_i = i + move.first;
   int next_j = j + move.second;
}

```